client/src/components/History/Content/GenericItem.vue

```diff
@@ -76,17 +78,38 @@ export default {
     },
     methods: {
         ...mapActions(useHistoryStore, ["applyFilters"]),
-        onDelete(item, recursive = false) {
-            deleteContent(item, { recursive: recursive });
+        async onDelete(item, recursive = false) {
+            try {
+                await deleteContent(item, { recursive: recursive });
+            } catch (error) {
+                this.onError(error, "Failed to delete item");
+            }
         },
-        onUndelete(item) {
-            updateContentFields(item, { deleted: false });
+        onError(e, title = "Error") {
+            const error = errorMessageAsString(e, "Dataset operation failed.");
+            Toast.error(error, title);
+            console.error(error);
         },
-        onHide(item) {
-            updateContentFields(item, { visible: false });
+        async onUndelete(item) {
+            try {
+                await updateContentFields(item, { deleted: false });
+            } catch (error) {
+                this.onError(error, "Failed to undelete item");
+            }
         },
-        onUnhide(item) {
-            updateContentFields(item, { visible: true });
+        async onHide(item) {
+            try {
+                await updateContentFields(item, { visible: false });
+            } catch (error) {
+                this.onError(error, "Failed to hide item");
+            }
+        },
+        async onUnhide(item) {
+            try {
+                await updateContentFields(item, { visible: true });
+            } catch (error) {
+                this.onError(error, "Failed to unhide item");
```

Comment:
Can you use `errorMessageAsString` so we get a more informative error for all the API operations ?

Reply:
If you look at the `onError` method, it uses `errorMessageAsString`) there:
https://github.com/galaxyproject/galaxy/pull/18203/files#diff-df66bdef0323b531878433090f832875b4aeb6f0801792846ebb2621b40604b6R88-L84

Reply:
ok, cool, thank you!